lightning/src/util/config.rs

```diff
@@ -126,6 +126,26 @@ pub struct ChannelHandshakeConfig {
 	///
 	/// [`KeysInterface::get_shutdown_scriptpubkey`]: crate::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey
 	pub commit_upfront_shutdown_pubkey: bool,
+
+	/// `their_channel_reserve_satoshis` is the minimum amount that the other node is to keep as a direct payment.
+	/// This ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain
+	///
+	/// `their_channel_reserve_proportion_millionth` is for us to configure counterparty's channel reserve
+	/// for both outgoing and incoming channels.
+	///
+	/// channel_reserve values greater than 30% could be considered highly unreasonable, since that
+	/// amount can never be used for payments.
+	/// Also, if our selected channel_reserve for counterparty and counterparty's selected channel_reserve
+	/// for us, sum upto or greater than channel_value, channel negotiations will fail.
+	///
+	/// `their_channel_reserve_proportion_millionth` is millionths of a channel_value to use as `their_channel_reserve_satoshis`.
+	///
+	/// Default Value is 1% of channel_value, configured as 10,000 millionth
+	/// Lower Bound on `their_channel_reserve_satoshis` is 1000 sats provided as safe implementation limit.
+	/// Max `their_channel_reserve_satoshis` can be equal to channel_value i.e 1000,000 millionth
```

Comment:
I find it very strange that the range is upper bound to value that is guaranteed to fail. If that's the case, then why is it even upper bound at all?
If we want to have an upper bound, wouldn't we at least want make the bound a value that's more likely to succeed?

Reply:
yes the idea was to handle unreasonable reserve configs by docs, because technically even a 99.99% reserve might work. and 60-30 channel_reserve b/w parties is unreasonable but valid.

Reply:
Ok I still find it a bit strange to bound the value to an upper bound that's guaranteed to fail, instead of just not having any bound at all and just let the channel negotiations fail if the user sets a too high value. But if others are ok with it, I am as well :).